crates/kitsune_p2p/dht_arc/src/dht_arc.rs

```diff
@@ -39,13 +39,30 @@ pub mod gaps;
 /// a u32 dht arc
 pub struct DhtLocation(pub Wrapping<u32>);
 
+impl DhtLocation {
+    pub fn new(loc: u32) -> Self {
+        Self(Wrapping(loc))
+    }
+
+    pub fn as_u32(&self) -> u32 {
```

Comment:
Putting on my pedantic hat for a moment, `as_` fns return references... this returns an owned type, so should be `to_`

```suggestion
    pub fn to_u32(&self) -> u32 {
```